src/Altinn.Platform/Altinn.Platform.Authorization/Authorization/Controllers/DelegationsController.cs

```diff
@@ -82,6 +85,66 @@ public async Task<ActionResult> Post([FromBody] List<Rule> rules)
             }
         }
 
+        /// <summary>
+        /// Endpoint for retrieving delegated rules between parties
+        /// </summary>
+        /// <response code="400">Bad Request</response>
+        /// <response code="500">Internal Server Error</response>
+        [HttpPost]
+        [Authorize(Policy = AuthzConstants.ALTINNII_AUTHORIZATION)]
+        [Route("authorization/api/v1/[controller]/GetRules")]
+        public async Task<ActionResult<List<Rule>>> GetRules([FromBody] RuleQuery ruleQuery, [FromQuery] bool onlyDirectDelegations = false)
+        {
+            List<int> coveredByPartyIds = new List<int>();
+            List<int> coveredByUserIds = new List<int>();
+            List<int> offeredByPartyIds = new List<int>();
+            List<string> orgApps = new List<string>();
+
+            foreach (PolicyMatch policyMatch in ruleQuery.PolicyMatches)
+            {
+                string org = policyMatch.Resource.FirstOrDefault(match => match.Id == XacmlRequestAttribute.OrgAttribute)?.Value;
+                string app = policyMatch.Resource.FirstOrDefault(match => match.Id == XacmlRequestAttribute.AppAttribute)?.Value;
+                if (!string.IsNullOrEmpty(org) && !string.IsNullOrEmpty(app))
+                {
+                    orgApps.Add($"{org}/{app}");
+                }
+
+                if (DelegationHelper.TryGetCoveredByPartyIdFromMatch(policyMatch.CoveredBy, out int partyId))
+                {
+                    coveredByPartyIds.Add(partyId);
+                }
+                else if (DelegationHelper.TryGetCoveredByUserIdFromMatch(policyMatch.CoveredBy, out int userId))
+                {
+                    coveredByUserIds.Add(userId);
+                }
+
+                if (ruleQuery.KeyRolePartyIds.Any(id => id != 0))
+                {
+                    coveredByPartyIds.AddRange(ruleQuery.KeyRolePartyIds);
+                }
+
+                if (ruleQuery.ParentPartyId != 0)
+                {
+                    offeredByPartyIds.Add(ruleQuery.ParentPartyId);
+                }
+
+                if (policyMatch.OfferedByPartyId != 0)
+                {
+                    offeredByPartyIds.Add(policyMatch.OfferedByPartyId);
+                }
+            }
+
+            try
+            {
+                return Ok(await _pip.GetRulesAsync(orgApps, offeredByPartyIds, coveredByPartyIds, coveredByUserIds));
+            }
+            catch (Exception e)
+            {
+                _logger.LogError($"Unable to get rules. {e}");
```

Comment:
Actually. consider removing the try catch. Logging is performed by the Web API request pipeline if exceptions are allowed to pass through. The same mechanisms will produce a Http Response will status code 500.

Reply:
Use try catch for situations where we actually handle the exception.

Reply:
Agree. As it is implemented this can be just removed and left for the exceptionhandler middleware to deal with. However, if we at this time have useful information that is not available in the exception (ie. app list, parties involved), we should log this (using [templates for structured logging](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/?view=aspnetcore-6.0#log-message-template)) and just `throw;` again.

Reply:
Removed the unnecessary try-catch.